schema/anyof_test.go

```diff
@@ -86,3 +86,32 @@ func TestAnyOfValidate(t *testing.T) {
 		cases[i].Run(t)
 	}
 }
+
+func TestAnyOfSerialize(t *testing.T) {
+	cases := []fieldValidatorTestCase{
+		{
+			Name:       "{Bool,Bool}.Serialize(true)",
+			Validator:  schema.AnyOf{&schema.Bool{}, &schema.Bool{}},
+			Serializer: schema.AnyOf{&schema.Bool{}, &schema.Bool{}},
+			Input:      true,
+			Expect:     true,
+		},
+		{
+			Name:       `{Bool,IP}.Serialize("1.2.3.4")`,
+			Validator:  schema.AnyOf{&schema.Bool{}, &schema.IP{}},
+			Serializer: schema.AnyOf{&schema.Bool{}, &schema.IP{}},
+			Input:      "1.2.3.4",
+			Expect:     "1.2.3.4",
+		},
+		{
+			Name:       `{Bool,IP{StoreBinary:true}}.Serialize("1.2.3.4")`,
+			Validator:  schema.AnyOf{&schema.Bool{}, &schema.IP{}},
+			Serializer: schema.AnyOf{&schema.Bool{}, &schema.IP{StoreBinary: true}},
+			Input:      "1.2.3.4",
```

Comment:
byte array as input

Reply:
Also, must set `StoreBinary` to `true`

Reply:
If possible, implement a dummy Serializer as well so we have a test with two serializers. E.g.

```
Validator: schema.AnyOf{&hexByteArray{},&schema.IP{StoreBinary:true}}
Input: [1, 2, 3, 4]
Output:  "0x1234"
Validator: schema.AnyOf{&schema.IP{StoreBinary:true},& hexByteArray{}}
Input: [1, 2, 3, 4]
Output:  "1.2.3.4"
```

This test would also clearly document the current behaviour in terms of picking a serializer.

We should also have an example with a failing serializer to see that it can pick the right one when the unput types are differen. E.g. 

```
Validator: schema.AnyOf{&intString{},&schema.IP{StoreBinary:true}}
Input: [1, 2, 3, 4]
Output:  "1.2.3.4"
Validator: schema.AnyOf{&schema.IP{StoreBinary:true},&intString{}}
Input: [1, 2, 3, 4]
Output:  "1.2.3.4"
Validator: schema.AnyOf{&intString{},&schema.IP{StoreBinary:true}}
Input: int64(100)
Output:  "100"
Validator: schema.AnyOf{&schema.IP{StoreBinary:true},&intString{}}
Input: int64(100)
Output:  "100"
```